runtime/autoload/provider/ruby.vim

```diff
@@ -16,19 +22,46 @@ function! provider#ruby#Require(host) abort
 
   try
     let channel_id = rpcstart(provider#ruby#Prog(), args)
-
-    if rpcrequest(channel_id, 'poll') == 'ok'
+    if rpcrequest(channel_id, 'poll') ==# 'ok'
       return channel_id
     endif
   catch
     echomsg v:throwpoint
     echomsg v:exception
   endtry
-
-  throw remote#host#LoadErrorForHost(a:host.orig_name,
-        \ '$NVIM_RUBY_LOG_FILE')
+  throw remote#host#LoadErrorForHost(a:host.orig_name, '$NVIM_RUBY_LOG_FILE')
 endfunction
 
-function! provider#ruby#Prog() abort
-  return 'neovim-ruby-host'
+function! provider#ruby#Call(method, args)
+  if s:err != ''
+    echoerr s:err
+    return
+  endif
+
+  if !exists('s:host')
+    let s:rpcrequest = function('rpcrequest')
```

Comment:
well it's called as a "closure" down on 54 - but I'm not sure why it does that.  I think I may have written this long ago - `s:host` needs to be captured, but I don't know why we need a different handle on the rpcrequest function.